UT4MasterServer.Services/Scoped/RatingsService.cs

```diff
@@ -0,0 +1,210 @@
+﻿using MongoDB.Driver;
+using UT4MasterServer.Common;
+using UT4MasterServer.Models;
+using UT4MasterServer.Models.DTO.Response;
+using UT4MasterServer.Models.DTO.Responses;
+
+namespace UT4MasterServer.Services;
+
+public sealed class RatingsService
+{
+	private readonly IMongoCollection<Rating> ratingsCollection;
+
+	public RatingsService(DatabaseContext dbContext)
+	{
+		ratingsCollection = dbContext.Database.GetCollection<Rating>("ratings");
+	}
+
+	public async Task<MMRBulkResponse> GetRatingsAsync(EpicID accountID, MMRBulkResponse mmrBulk)
+	{
+		var ratingTypes = mmrBulk.RatingTypes.Intersect(Rating.AllowedRatingTypes);
+		var filter = Builders<Rating>.Filter.Eq(f => f.AccountID, accountID) &
+					 Builders<Rating>.Filter.In(f => f.RatingType, ratingTypes);
+		var ratings = await ratingsCollection.Find(filter).ToListAsync();
+
+		var result = new MMRBulkResponse();
+
+		foreach (var ratingType in ratingTypes)
+		{
+			var rating = ratings.Where(w => w.RatingType == ratingType).FirstOrDefault();
+
+			result.RatingTypes.Add(ratingType);
+			result.Ratings.Add(rating?.RatingValue / Rating.Precision ?? Rating.DefaultRating);
+			result.NumGamesPlayed.Add(rating?.GamesPlayed ?? 0);
+		}
+
+		return result;
+	}
+
+	public async Task<MMRRatingResponse> GetRatingAsync(EpicID accountID, string ratingType)
+	{
+		var filter = Builders<Rating>.Filter.Eq(f => f.AccountID, accountID) &
+					 Builders<Rating>.Filter.Eq(f => f.RatingType, ratingType);
+		var rating = await ratingsCollection.Find(filter).FirstOrDefaultAsync();
+
+		var result = new MMRRatingResponse()
+		{
+			Rating = rating?.RatingValue / Rating.Precision ?? Rating.DefaultRating,
+			GamesPlayed = rating?.GamesPlayed ?? 0,
+		};
```

Comment:
do
```cs
if (rating == null)
{
	return new MMRRatingResponse();
}
```
so you don't have to do null checks later

Reply:
Done.